src/status_im/contexts/settings/wallet/events_test.cljs

```diff
@@ -18,3 +20,35 @@
                 (sut/rename-keypair cofx
                                     [{:key-uid      key-uid
                                       :keypair-name new-keypair-name}])))))
+
+(deftest wallet-validate-seed-phrase-test
+  (let [cofx               {:db {}}
+        seed-phrase-masked (security/mask-data "seed phrase")
+        on-success         #(prn "success")
+        on-error           #(prn "error")
+        expected           {:fx [[:multiaccount/validate-mnemonic
+                                  [seed-phrase-masked on-success on-error]]]}]
+    (testing "make-seed-phrase-fully-operable"
+      (is (= expected
+             (sut/wallet-validate-seed-phrase
+              cofx
+              [seed-phrase-masked on-success on-error]))))))
+
+(deftest make-seed-phrase-keypair-fully-operable-test
+  (let [cofx            {:db {}}
+        mnemonic        "seed phrase"
+        password        "password"
+        mnemonic-masked (security/mask-data mnemonic)
+        password-masked (security/mask-data password)
+        on-success      #(prn "success")
+        on-error        #(prn "error")
+        expected        {:fx [[:json-rpc/call
+                               [{:method     "accounts_makeSeedPhraseKeypairFullyOperable"
+                                 :params     [mnemonic (native-module/sha3 password)]
+                                 :on-success fn?
```

Comment:
Here wouldn't it be better to `match?` directly against `on-success` and `on-error` instead of `fn?` since you're passing the function instances directly.

Reply:
I had tried doing that, but I was running into an issue where the functions weren’t being compared but instead were being called as validators. I’ll double check if that’s happening, but I think `match?` uses functions as validators.